Name Resolution and Type Alias Expansion

In `tests/todo/append-bug-name.hs` the check for
`prop_app_nil_ok` works fine, but `prop_app_nil_bad` doesn't.

The only difference is the type of axiom_append_nil_{bad, ok},
in which the only difference is that one uses the {...N...} while
the other does not. For some reason this yields two **different** 
refinements, one with `Append.N` and one with `N`, which 
causes the problem. 

In short, there is some odd name resolution bug that interacts 
with the type alias expansion code. 

@spinda -- any idea whats going on?

It turns out that there is no well formed-ness check in the "assumed" types. 
If you remove the `assume` from the type and have 

```
{-@ axiom_append_nil_bad :: xs:L a -> Equal {append N xs} {xs} @-}
```

You will get the real name resolution error, that is 

```
 tests/todo/append-name-bug.hs:23:5: Error: Bad Type Specification
 Append.axiom_append_nil_bad :: xs:(L a) -> {v : Proof | append N xs == xs}
     Sort Error in Refinement: {v : Append.Proof | append N xs == xs}
     Unbound Symbol N
 Perhaps you meant: v
```

--- a/src/Language/Haskell/Liquid/Bare/OfType.hs
+++ b/src/Language/Haskell/Liquid/Bare/OfType.hs
@@ -186,18 +186,16 @@ failRTAliasApp l rta _ _
 expandRTAliasApp :: SourcePos -> RTAlias RTyVar SpecType -> [BareType] -> RReft -> BareM SpecType
 expandRTAliasApp l rta args r
   | length args == length αs + length εs
-    = do args' <- mapM (ofBareType l) args
-         let ts  = take (length αs) args'
-             αts = zipWith (\α t -> (α, toRSort t, t)) αs ts
-         return $ subst su . (`strengthen` r) . subsTyVars_meet αts $ rtBody rta
+    = do ts <- mapM (ofBareType l)                   $ take (length αs) args
+         es <- mapM (resolve l . exprArg (show err)) $ drop (length αs) args
+         let tsu = zipWith (\α t -> (α, toRSort t, t)) αs ts
+         let esu = mkSubst $ zip (symbol <$> εs) es
+         return $ subst esu . (`strengthen` r) . subsTyVars_meet tsu $ rtBody rta
   | otherwise
     = Ex.throw err
   where
-    su        = mkSubst $ zip (symbol <$> εs) es
     αs        = rtTArgs rta
     εs        = rtVArgs rta
-    es_       = drop (length αs) args
-    es        = map (exprArg $ show err) es_
     err       :: Error
     err       = ErrAliasApp (sourcePosSrcSpan l) (length args) (pprint $ rtName rta) (sourcePosSrcSpan $ rtPos rta) (length αs + length εs)
 

